rpcserver.go

```diff
@@ -1461,13 +1473,35 @@ func (r *rpcServer) SendPayment(paymentStream lnrpc.Lightning_SendPaymentServer)
 
 					// TODO(roasbeef): eliminate necessary
 					// encode/decode
-					nextPayment.Dest = payReq.Destination.SerializeCompressed()
-					nextPayment.Amt = int64(payReq.Amount)
-					nextPayment.PaymentHash = payReq.PaymentHash[:]
+					p.dest = payReq.Destination.SerializeCompressed()
+
+					if payReq.MilliSat == nil {
+						err := fmt.Errorf("only payment" +
+							" requests specifying" +
+							" the amount are" +
+							" currently supported")
+						select {
+						case errChan <- err:
+						case <-reqQuit:
+							return
+						}
+						return
+					}
+					p.msat = *payReq.MilliSat
+					p.pHash = payReq.PaymentHash[:]
+				} else {
```

Comment:
When sending (both here and within `SendPaymentSync`), if a full invoice was passed in, then the expiration of the invoice _is not_ being properly observed. If the payment request has expired, then the payment attempt should be rejected.

Reply:
Added a check for this. 

While doing this I realized `time.Time` is not the best way of expressing timespans in Go, so I had to cast `payReq.Expiry` to seconds, and then to `time.Duration`. We might consider changing the `zpay32.Invoice` struct to express `Expiry` directly in `time.Duration`. I will also consider moving the "default to 3600 seconds logic" within `zpay32` in the same change.